File: spring-todomvc-rest/src/main/java/org/example/todomvc/todo/TodoController.java
package org.example.todomvc.todo;

import java.util.List;

import javax.ws.rs.PathParam;

import org.example.todomvc.ApiRequestMapping;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@ApiRequestMapping
public class TodoController {
	@Autowired
	private TodoRepository todoRepository;
	
	@GetMapping(path="/todo/{id}")
	public List<Todo> findByUser(@PathParam("id") Integer assigneeId) {
		return todoRepository.findByAssignee(assigneeId);
	}
	
	@GetMapping(path="/todoTypes")
	public List<TodoType> listTodoTypes() {
		return todoRepository.listTodoTypes();
	}

	@GetMapping(path="/greet")
	public String greet() {
		return "Hello world!";
	}

}

File: htmx/src/main/kotlin/com/example/spring/htmx/api/data/TodoController.kt
package com.example.spring.htmx.api.data

import com.example.spring.htmx.service.TodoService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.ui.ModelMap
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/todos")
class TodoController {

  @Autowired
  lateinit var todoService: TodoService

  @GetMapping("")
  fun todos(model: ModelMap) = todoService.todos

  @GetMapping("{id}")
  fun todo(@PathVariable id: String, model: ModelMap) = todoService.todos.find { it.id == id }

}